### PR TITLE
Add TimeStamp support.

### PR DESCRIPTION
**Pull request by VelvetThePanda**
# Summary
Adds support for timestamps, documented in discord/discord-api-docs#3135.

### CODE REVIEW

**Comment by IDoEverything** [comment]
Shouldn't there be support for all the flags?

**Comment by VelvetThePanda** [comment]
*Technically* speaking, yes, but:
1: that would require adding an enum, I suppose
2: Embed timestamps also use relative, which imo is the most useful

~~3: Since you're here anyway do you mind triaging this.~~

**Comment by IDoEverything** [comment]
Doesn't really make sense to restrict it to only the relative one. Nothing wrong with an enum either

**Comment by VelvetThePanda** [comment]
@IDoEverything done.

**Review by IDoEverything — APPROVED** [review]

File: DSharpPlus/Enums/TimestampFormat.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+namespace DSharpPlus
+{
+    /// <summary>
+    /// Denotes the type of formatting to use for timestamps.
+    /// </summary>
+    public enum TimestampFormat
```
Comment (Emzi0767) [inline DSharpPlus/Enums/TimestampFormat.cs:28]: `: char`?

File: DSharpPlus/Formatter.cs [hunk 2]
```diff
@@ -53,6 +53,29 @@ public static string BlockCode(string content, string language = "")
         public static string InlineCode(string content)
             => $"`{content}`";
 
+        /// <summary>
+        /// Creates a rendered timestamp.
+        /// </summary>
+        /// <param name="time">The time from now.</param>
+        /// <param name="format">The format to render the timestamp in. Defaults to relative.</param>
+        /// <returns>A formatted timestamp relative to now.</returns>
+        public static string TimeStamp(TimeSpan time, TimestampFormat format = TimestampFormat.RelativeTime)
```
Comment (Emzi0767) [inline DSharpPlus/Formatter.cs:62]: `s/S/s/`

Comment (VelvetThePanda) [reply DSharpPlus/Formatter.cs:62]: what?

Comment (VelvetThePanda) [reply DSharpPlus/Formatter.cs:62]: oh, sed

Comment (Emzi0767) [reply DSharpPlus/Formatter.cs:62]: Yes, also I forgot

**Review by Emzi0767 — CHANGES_REQUESTED** [review]
This is what I forgot

**Comment by VelvetThePanda** [comment]
@Emzi0767 should be good now

**Review by Emzi0767 — APPROVED** [review]

**Pull request merged**